helix-core/src/main/java/org/apache/helix/model/InstanceConfig.java

```diff
@@ -704,7 +706,7 @@ public static InstanceConfig toInstanceConfig(String instanceId) {
 
     }
 
-    config.setInstanceEnabled(true);
+    config.setInstanceEnabled(HELIX_ENABLED_DEFAULT_VALUE);
```

Comment:
Let's not add this. Helix basic principal is minimized the writes to ZNRecord. If it does not show in the ZNRecord, usually we use default value in get method but not explicit add it to record.